Neos.Fusion/Classes/FusionObjects/AttributesImplementation.php

```diff
@@ -33,34 +33,14 @@ class AttributesImplementation extends AbstractArrayFusionObject
     public function evaluate()
     {
         $allowEmpty = $this->getAllowEmpty();
-
-        $renderedAttributes = '';
+        $attributes = [];
         foreach (array_keys($this->properties) as $attributeName) {
             if ($attributeName === '__meta' || in_array($attributeName, $this->ignoreProperties)) {
                 continue;
             }
-
-            $encodedAttributeName = htmlspecialchars($attributeName, ENT_COMPAT, 'UTF-8', false);
-            $attributeValue = $this->fusionValue($attributeName);
-            if ($attributeValue === null || $attributeValue === false) {
-                // No op
-            } elseif ($attributeValue === true || $attributeValue === '') {
-                $renderedAttributes .= ' ' . $encodedAttributeName . ($allowEmpty ? '' : '=""');
-            } else {
-                if (is_array($attributeValue)) {
-                    $joinedAttributeValue = '';
-                    foreach ($attributeValue as $attributeValuePart) {
-                        if ((string)$attributeValuePart !== '') {
-                            $joinedAttributeValue .= ' ' . trim($attributeValuePart);
-                        }
-                    }
-                    $attributeValue = trim($joinedAttributeValue);
-                }
-                $encodedAttributeValue = htmlspecialchars($attributeValue, ENT_COMPAT, 'UTF-8', false);
-                $renderedAttributes .= ' ' . $encodedAttributeName . '="' . $encodedAttributeValue . '"';
-            }
+            $attributes[$attributeName] = $this->fusionValue($attributeName);
         }
-        return $renderedAttributes;
+        return TagImplementation::renderAttributes($attributes, $allowEmpty);
```

Comment:
I stumbled upon the fact that this fusion object uses another implementation to render.. But this is just for backwards compatibility, right?
If so, we should add a `@deprecated` annotation IMO

Reply:
Yes i wanted to render attributes always via the same function but also did'nt want to introduce an AttributeRenderingService.

Neos.Fusion:Attributes has to stay for a while as many Fluid-templates actually pass attributes that way. That is why i did'nt mark it deprecated. I think this will stay as long as fluid.

Reply:
I see your reasoning to introduce the central function (even though I really despise public static methods *g).
And I also think that `Neos.Fusion:Attributes` will stick around for some time.
But IMO we should go for a unique way and if I'm right there is no reason not to use `Neos.Fusion:DataStructure` for attributes.
So my personal opinion would be: Let's go "all in", deprecate `Neos.Fusion:Attributes` and use `Neos.Fusion:DataStructure` in the core. (and I'm happy to take care of replacing it in Core and Demo site etc)

Reply:
for the record: This would be just a slight preference, I'm still happy with the change as is - it improves things!

Reply:
@bwaidelich, i agree. It is easier to live with two methods until Attributes is finally removed. I will adjust the PR.

Do you see a valid usecase for still supporting attributes.@allowEmpty. Only case I can think of is xml generation. But would that be done with Neos.Fusion:Tag?

Reply:
Great, thanks!

> Do you see a valid usecase for still supporting attributes.@allowEmpty

No, I don't. if it is true by default (which it is) you can always "simulate" the other case by chaning `true/''` to `'true'` in the attributes as far as I can tell